beamer/chain.py

```diff
@@ -318,11 +318,24 @@ def fill_request(request: Request, context: Context) -> None:
         )
         return
 
+    allowance = context.config.token_checker.allowance(request.target_chain_id, token.address)
+    if allowance is None:
+        allowance = request.amount
+
+    if allowance < request.amount and allowance != -1:
+        context.logger.info(
+            "Allowance for token smaller than requested amount",
+            token=token.address,
+            allowance=allowance,
+            request_amount=request.amount,
+        )
+        return
+
     if (
         token.functions.allowance(context.address, context.fill_manager.address).call()
         < request.amount
     ):
-        func = token.functions.approve(context.fill_manager.address, request.amount)
+        func = token.functions.approve(context.fill_manager.address, allowance)
```

Comment:
Just to check whether my understanding is correct: if `allowance` is `-1`, when preparing tx data it will be converted internally by `web3` to the maximum representable value of `uint256` (`0xff...ff`), which is what we want?

Reply:
good idea! unfortunately, web3.py cannot do the special casing. But moving this into token_checker.allowance is the right thing to do.

Reply:
so actually we could remove the comparison to -1 above.